File: app/src/main/java/com/jeksonshar/jweather/repository/room/RoomRepository.java
package com.jeksonshar.jweather.repository.room;

import android.content.Context;

import androidx.room.Room;

import com.jeksonshar.jweather.model.WeatherModel;
import com.jeksonshar.jweather.repository.Repository;

import java.util.ArrayList;
import java.util.List;

public class RoomRepository implements Repository {

    private WeatherDao mWeatherDao;

    public RoomRepository(Context context) {
        mWeatherDao = Room
                .databaseBuilder(context, WeatherDataBase.class, "weather-database4.sqLite")
                .allowMainThreadQueries()
                .build()
                .getWeatherDao();
    }

    @Override
    public List<WeatherModel> getAllItemsWeather() {
        List<WeatherEntity> weatherEntities = mWeatherDao.getAllItemsWeather();
        List<WeatherModel> weatherModels = new ArrayList<>();

        for (WeatherEntity weatherEntity: weatherEntities) {
            weatherModels.add(Converter.convert(weatherEntity));
        }
        return weatherModels;
    }

    @Override
    public void saveItemsInRoom(List<WeatherModel> weatherModels) {
        List<WeatherEntity> weatherEntities = new ArrayList<>();

        for (WeatherModel weatherModel: weatherModels) {
            weatherEntities.add(Converter.convert(weatherModel));
        }

        if (mWeatherDao.getAllItemsWeather().isEmpty()) {
            for (WeatherEntity weatherEntity: weatherEntities) {
                mWeatherDao.saveWeatherModel(weatherEntity);
            }
        } else {
            for (WeatherEntity weatherEntity : weatherEntities) {
                mWeatherDao.updateWeatherModel(weatherEntity);
            }
        }
    }
//    @Override
//    public WeatherModel getItemWeatherByID(int id) {
//        return ;
//    }
//
//    @Override
//    public void update(WeatherModel weatherModel) { // ??? is need to Jweather?
//
//    }
}

File: app/src/main/java/com/jeksonshar/jweather/repository/room/RoomRepository.kt
package com.jeksonshar.jweather.repository.room

import android.content.Context
import com.jeksonshar.jweather.model.WeatherModel
import androidx.room.Room
import com.jeksonshar.jweather.repository.Repository
import java.util.ArrayList

class RoomRepository(context: Context?) : Repository { //TODO контекст NotNull

    private val mWeatherDao: WeatherDao? = Room
            .databaseBuilder(context!!, WeatherDataBase::class.java, "weather-database5.sqLite")
            .allowMainThreadQueries() //TODO перенести на бекграунд поток
            .build() //TODO dao и бд отдельные переменные
            .weatherDao

    override val allItemsWeather: List<WeatherModel>
        get() {
            val weatherEntities = mWeatherDao?.allItemsWeather
            val weatherModels: MutableList<WeatherModel> = ArrayList()

            for (weatherEntity in weatherEntities!!) { //TODO убрать восклицательные знаки
                Converter.convert(weatherEntity)?.let { weatherModels.add(it) }
            }
            return weatherModels
        }

    override fun saveItemsInRoom(weatherModels: List<WeatherModel>?) {

        val weatherEntities: MutableList<WeatherEntity> = ArrayList()

        for (weatherModel in weatherModels!!) { //TODO убрать восклицательные знаки
            weatherEntities.add(Converter.convert(weatherModel))
        }

        if (mWeatherDao?.allItemsWeather!!.isEmpty()) { //TODO убрать восклицательные знаки
            for (weatherEntity in weatherEntities) {
                mWeatherDao.saveWeatherModel(weatherEntity)
            }
        } else {
            for (weatherEntity in weatherEntities) {
                mWeatherDao.updateWeatherModel(weatherEntity)
            }
        }
    }
}